packages/palette/src/elements/Shelf/ShelfScrollBar.tsx

```diff
@@ -0,0 +1,207 @@
+          ((event.pageX - startX.current) * scrollWidth) / trackWidth
+
+        viewport.scrollLeft = offsetX.current + delta
+      }
+
+      thumb.addEventListener("mousedown", handleMouseDown)
+      document.addEventListener("mousemove", handleMouseMove)
+      document.addEventListener("mouseup", handleMouseUp)
+
+      return () => {
+        thumb.removeEventListener("mousedown", handleMouseDown)
+        document.removeEventListener("mousemove", handleMouseMove)
+        document.removeEventListener("mouseup", handleMouseUp)
+      }
+    }, [viewport, clientWidth, scrollWidth, scrollLeft, trackWidth])
+
+    // Handle click on track scrolling
+    useEffect(() => {
+      if (!trackRef.current) return
+
+      const { current: track } = trackRef
+      const { current: thumb } = thumbRef
+
+      const handleMouseDown = (event: MouseEvent) => {
+        const x =
+          event.pageX -
+          track.getBoundingClientRect().left -
+          thumb.clientWidth / 2
+
+        const delta = (x * scrollWidth) / trackWidth
+
+        viewport.scrollLeft = delta
+      }
+
+      track.addEventListener("mousedown", handleMouseDown)
+      return () => {
+        track.removeEventListener("mousedown", handleMouseDown)
+      }
+    }, [viewport, scrollWidth, trackWidth])
+
+    return (
+      <>
+        {/* Track */}
+        <Box
+          ref={trackRef as any}
+          position="relative"
+          height={3}
+          bg="black15"
+          width="100%"
+          role="scrollbar"
+          aria-orientation="vertical"
+          aria-valuemax={100}
+          aria-valuemin={0}
+          aria-valuenow={progress}
+          {...rest}
+        >
+          {/* Pad out hit area. Click events will propagate to underlying trackRef */}
+          <TrackHitArea />
+
+          {/* Thumb */}
+          <Box
+            position="relative"
+            bg="black60"
+            height="100%"
+            borderRadius={3}
+            width={normalizedThumbWidth}
+            style={{
+              transform: `translateX(${normalizedThumbOffset}px)`,
+              backfaceVisibility: "hidden",
+            }}
+          >
+            <HitArea ref={thumbRef as any} tabIndex={-1} />
+          </Box>
+        </Box>
+      </>
+    )
+  }
+)
+
+const TrackHitArea = styled(Box)`
+  position: absolute;
+  top: -10px;
+  bottom: -10px;
+  right: 0;
+  left: 0;
+  width: 100%;
+`
+
+const HitArea = styled(Clickable)`
+  position: absolute;
+  top: -10px;
+  bottom: -10px;
+  width: 100%;
+  left: 0;
+  opacity: 0;
+  transition: opacity 250ms;
+
+  &:hover {
+    opacity: 1;
+  }
+
+  &:after {
+    content: "";
+    display: block;
+    position: absolute;
+    left: 0;
+    width: 100%;
+    background-color: ${themeGet("colors.black100")};
+    /*
+     * Positioning this using top/bottom and translate each have cross-browser
+     * issues in Safari. Using a hard-coded height/negative margin works on all browsers.
+     */
```

Comment:
Shocked that this was the one cross-browser issue I ran into. Safari apparently handles positioning slightly differently than Chrome/Firefox when it comes to negative values? I should do a minimal replication to better understand what's happening.